objax/module.py

```diff
@@ -53,6 +53,54 @@ def __call__(self, *args, **kwargs):
         raise NotImplementedError
 
 
+class ForceArgs(Module):
+    """Forces override of arguments of given module."""
+
+    @staticmethod
+    def undo(module: Module):
+        """Undo ForceArgs on each submodule of the module.
+
+        Args:
+            module: module for which to undo ForceArgs.
+        """
+        if isinstance(module, ForceArgs):
+            raise ValueError('ForceArgs.undo can not be called on ForceArgs, call it on parent module instead.')
+        if isinstance(module, ModuleList):
+            for idx, v in enumerate(module):
+                if isinstance(v, Module):
+                    if isinstance(v, ForceArgs):
+                        v = v._wrapped
+                        module[idx] = v
+                    ForceArgs.undo(v)
+        else:
+            for k, v in module.__dict__.items():
+                if isinstance(v, Module):
+                    if isinstance(v, ForceArgs):
+                        v = v._wrapped
+                        module.__dict__[k] = v
+                    ForceArgs.undo(v)
+
+    def __init__(self, module: Module, **kwargs):
+        """Initializes ForceArgs.
+
+        Args:
+            module: base module, which argument will be overridden.
+            kwargs: values of keyword arguments which will be forced to use.
+        """
+        if isinstance(module, ForceArgs):
+            raise ValueError('Can not force arguments on ForceArgs module.')
+        self._wrapped = module
+        self.forced_kwargs = kwargs
```

Comment:
Two things:
1. `base_module` can be simply called `module`.
2. Rename `_wrapped` to `__wrapped__` which is used by python to determine the signature of a function.
```suggestion
    def __init__(self, module: Module, **kwargs):
        """Initializes ForceArgs.

        Args:
            module: module which arguments will be overridden.
            kwargs: values of keyword arguments which will be forced to use.
        """
        if isinstance(module, ForceArgs):
            raise ValueError('Can not force arguments on ForceArgs module.')
        self.__wrapped__ = module
        self.forced_kwargs = kwargs
```

Reply:
Done